safekeeper/src/timeline_manager.rs

```diff
@@ -579,18 +602,43 @@ impl Manager {
         }
     }
 
+    /// Reset partial backup state and remove its remote storage data. Since it
+    /// might concurrently uploading something, cancel the task first.
+    async fn backup_partial_reset(&mut self) -> anyhow::Result<Vec<String>> {
+        info!("resetting partial backup state");
+        // Force unevict timeline if it is evicted before erasing partial backup
+        // state. The intended use of this function is to drop corrupted remote
+        // state; we haven't enabled local files deletion yet anywhere,
+        // so direct switch is safe.
+        if self.is_offloaded {
+            self.tli.switch_to_present().await?;
+            // switch manager state as soon as possible
+            self.is_offloaded = false;
+        }
+
+        if let Some((handle, cancel)) = &mut self.partial_backup_task {
+            cancel.cancel();
+            info!("cancelled partial backup task, awaiting it");
+            let res = handle.await;
```

Comment:
FYI in `wal_backup_partial.rs` this line can take quite some time to finish:
```
let _upload_permit = limiter.acquire_partial_backup().await;
```

If many timelines want to upload partial segments at the same time, they will be congested around the `acquire_partial_backup()` rate limiter. I think we should add a select! with cancellation token there.

Reply:
yeah, fixed